web/src/ResourceStatusSummary.tsx

```diff
@@ -286,58 +287,87 @@ function ResourceMetadata(props: { counts: StatusCounts }) {
   return <></>
 }
 
-type ResourceStatusSummaryProps = {
-  view: Proto.webviewView
+/**
+ * The ResourceStatusSummary component takes a template type
```

Comment:
have you considered adding a `statuses: ResourceStatus[]` parameter instead?

i found all the type templating difficult to follow. are the type templates there mainly so you can get statuses from different object types, or are there other benefits?

Reply:
yup, that's why i added the template type! my thinking is that we want to be able to provide resource summaries throughout the app, but the resource data is going to come in different shapes. so this function accepts some resource type and a status getter function. but it sounds like that's creating level of abstraction that's difficult to follow, so i can simplify based on your suggestion.